packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -1,7 +1,61 @@
-import { AccountKeyDerivationPath, GroupedAddress, KeyRole } from '../types';
+import { AccountKeyDerivationPath, GroupedAddress } from '../types';
 import { Cardano } from '@cardano-sdk/core';
 import { isNotNil } from '@cardano-sdk/util';
 import uniq from 'lodash/uniq';
+import uniqBy from 'lodash/uniqBy';
+
+/**
+ * Gets whether any certificate in the provided certificate list requires a stake key signature.
+ *
+ * @param groupedAddresses The known grouped addresses.
+ * @param txBody The transaction body.
+ */
+// eslint-disable-next-line complexity
+const getStakingKeyPaths = (
+  groupedAddresses: GroupedAddress[],
+  txBody: Cardano.NewTxBodyAlonzo
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+) => {
+  const paths: Set<AccountKeyDerivationPath> = new Set();
+  for (const account of groupedAddresses) {
```

Comment:
uniqBy that I previously suggested is really an implementation detail of this function, so I'd hoist it here
```suggestion
  for (const account of uniq(groupedAddresses.map(({ rewardAccount }) => rewardAccount))) {
```